src/bootstrap/src/core/build_steps/vendor.rs

```diff
@@ -16,21 +17,32 @@ pub const VENDOR_DIR: &str = "vendor";
 /// Returns a `Vec` of `(path_to_manifest, submodules_required)` where
 /// `path_to_manifest` is the cargo workspace, and `submodules_required` is
 /// the set of submodules that must be available.
-pub fn default_paths_to_vendor(builder: &Builder<'_>) -> Vec<(PathBuf, Vec<&'static str>)> {
+pub fn default_paths_to_vendor(
+    builder: &Builder<'_>,
+    excluded_submodules: &BTreeSet<String>,
```

Comment:
It seems to me that all callers of `exclude_submodules` pass the same argument to it. Can we remove the argument and access `builder.config.dist_exclude_submodules_from_vendoring` directly inside of the function?